solution2.py

```diff
@@ -0,0 +1,42 @@
+# // Time Complexity : o(n*n + n/2*height => for loop inside addall at each leaf nodes) == o(n^2)
+# // Space Complexity : o(n*h)
```

Comment:
For every case, the Space will be O(n*n)/O(n^2)
A new list is created at every node in the tree!